src/app/component/eco-news/components/news-list/news-list.component.ts

```diff
@@ -55,46 +54,43 @@ export class NewsListComponent implements OnInit, OnDestroy {
       .subscribe((data) => this.isLoggedIn = data && data.userId);
   }
 
-  private fetchAllEcoNews(): void {
-    this.ecoNewsSubscription = this.ecoNewsService
-      .newsListSubject
-      .subscribe(this.setAllAndStartingElems.bind(this));
-  }
-
   public onScroll(): void {
     this.addElemsToCurrentList();
   }
 
   private addElemsToCurrentList(): void {
-    const loadingLength = this.allEcoNews.length - this.elements.length > 11 ? 11 :
-      this.allEcoNews.length - this.elements.length;
-
-    this.allEcoNews.forEach((element, index) => {
-      if (index >= this.iterator && index - this.iterator < loadingLength) {
-        this.elements[index] = element;
-      }
-    });
-
-    this.iterator = this.elements.length;
+    if (this.gridOutput) {
+      this.ecoNewsSubscription = this.ecoNewsService.getNewsListByTags(this.iterator++, 12, this.gridOutput)
+        .subscribe(this.setList.bind(this));
```

Comment:
The method 'bind' hasn't been used in Angular for a long time. Please check - is it really necessary?

Reply:
> The method 'bind' hasn't been used in Angular for a long time. Please check - is it really necessary?

It is necessary to pass the context to setList function